src/back/spv/writer.rs

```diff
@@ -765,17 +774,25 @@ impl Writer {
             }
             crate::TypeInner::Image { base, dim, flags } => {
                 let type_id = self.get_type_id(arena, base);
-                let dim = match dim {
-                    crate::ImageDimension::D1 => spirv::Dim::Dim1D,
-                    crate::ImageDimension::D2 => spirv::Dim::Dim2D,
-                    crate::ImageDimension::D3 => spirv::Dim::Dim2D,
-                    crate::ImageDimension::Cube => spirv::Dim::DimCube,
-                };
+                let dim = map_dim(dim);
                 self.try_add_capabilities(dim.required_capabilities());
 
-                instruction = self.instruction_type_image(id, type_id, dim, flags);
+                instruction = self.instruction_type_image(id, type_id, dim, flags, false);
                 self.lookup_type.insert(id, base);
             }
+            crate::TypeInner::DepthImage { dim, arrayed } => {
+                let type_id = 0; //TODO!
+                let dim = map_dim(dim);
+                self.try_add_capabilities(dim.required_capabilities());
+
+                let flags = if arrayed {
+                    crate::ImageFlags::ARRAYED
+                } else {
+                    crate::ImageFlags::empty()
+                };
+                instruction = self.instruction_type_image(id, type_id, dim, flags, true);
+                //self.lookup_type.insert(id, base);
```

Comment:
I am not that familiar with images yet, but how do we retrieve the handle here? Should that be a pointer to an image? I have read that you mentioned that this is determined at runtime for SPIR-V, so we are waiting for the Local value implementation am working on, right?

Reply:
I think it depends on the local type, yeah, since in know that the result is Float here but we don't have an ID to it.